src/cargo/core/registry.rs

```diff
@@ -303,13 +303,12 @@ http://doc.crates.io/specifying-dependencies.html#overriding-dependencies
             return Ok(())
         }
 
-        for id in real_deps {
+        if let Some(id) = real_deps.get(0) {
             let msg = format!("\
                 path override for crate `{}` has altered the original list of
                 dependencies; the dependency on `{}` was removed\n\n
                 {}", override_summary.package_id().name(), id.name(), boilerplate);
             self.source_config.config().shell().warn(&msg)?;
-            return Ok(())
```

Comment:
This is a semantic change?

Reply:
The loop never actually loops as the `return Ok(())` is always executed after the first item; if there is no first item, the next line also return `Ok(())`

Reply:
Let's at least stick to having a return here